### PR TITLE
Add DST support for naughtiness through Insight

### PR DESCRIPTION
**Pull request by penguin0616**

### CODE REVIEW

File: modmain.lua [hunk 1]
```diff
@@ -464,16 +476,24 @@ local function StatusPostConstruct(self)
...
+				self.naughtybadge.head:Hide()
+				self.naughtybadge.real_head = self.naughtybadge.icon:AddChild(UIAnim())
+			else
+				self.naughtybadge.real_head = self.naughtybadge.head
+			end
+			self.naughtybadge.real_head:GetAnimState():SetBank('krampus')
+			self.naughtybadge.real_head:GetAnimState():SetBuild('krampus_build')
+			self.naughtybadge.real_head:GetAnimState():SetPercent('hit', 1)
+			self.naughtybadge.real_head:SetScale(0.1)
+			self.naughtybadge.real_head:SetPosition(0, -32)
```
Comment (rezecib) [inline modmain.lua]: This looks like a no-op, what was the reasoning here?

File: modmain.lua [hunk 2]
```diff
@@ -42,6 +42,13 @@ local COMPACTSEASONS = SEASONOPTIONS == "Compact"
 local MICROSEASONS = SEASONOPTIONS == "Micro"
 local HUDSCALEFACTOR = GetModConfigData("HUDSCALEFACTOR")*.01
 
+if SHOWNAUGHTINESS then
+	if DST and not GLOBAL.KnownModIndex:IsModEnabled("workshop-2189004162") then -- shouldnt show naughtiness if this isnt here to provide it
+		SHOWNAUGHTINESS = false
+	end
+end
+	
+
```
Comment (rezecib) [inline modmain.lua]: Can you roll this into the CHECK_MODS section below? It'll make it easier to extend this if there are other mods that add naughtiness, and also checks for it two ways. If I remember correctly, KnownModIndex:IsModEnabled only gets mods that are already loaded.

File: modmain.lua [hunk 3]
```diff
@@ -437,7 +445,7 @@ local function KrampedPostInit(self)
 		self.inst:PushEvent("naughtydelta")
 	end
 end
-if SHOWNAUGHTINESS then
+if SHOWNAUGHTINESS and DST == false then -- only activate in DS
```
Comment (rezecib) [inline modmain.lua]: I'd prefer `not DST` and `DST` in place of `DST == false` and `DST == true`

Testing against exact boolean values like this implies that this variable should not be treated like a boolean but rather stores separate information for boolean and non-boolean values (such as nil).

(please update other places too)

File: modmain.lua [hunk 4]
```diff
@@ -453,9 +461,14 @@ local function StatusPostConstruct(self)
...
-			self.naughtiness.num:SetString(	(self.owner.components.kramped.actions or 0) .. "/" ..
-											(self.owner.components.kramped.threshold or 0) 			)
+		local function UpdateNaughty(_, data) -- player, data
+			if DST == false then
+				self.naughtiness.num:SetString(	(self.owner.components.kramped.actions or 0) .. "/" ..
+												(self.owner.components.kramped.threshold or 0) 			)
+			else
+				-- data = { actions = n, threshold = n }
+				self.naughtiness.num:SetString(data.actions .. "/" .. data.threshold)
+			end
```
Comment (rezecib) [inline modmain.lua]: I'd suggest adding a guard on this, mainly to defend against other mods potentially pushing this event:

```suggestion
			if DST then
				data = type(data) == "table" and data or {}
			else
				data = self.owner.components.kramped
			end
			local actions = type(data.actions) == "number" and data.actions or 0
			local threshold = type(data.threshold) == "number" and data.threshold or 0
			self.naughtiness.num:SetString(actions .. "/" .. threshold)
```

**Review by rezecib — CHANGES_REQUESTED** [review]

File: modmain.lua (hunk 2 above)
Comment (penguin0616) [reply modmain.lua]: Done. 
I think it only checks if they are enabled, but don't quote me on that.

File: modmain.lua (hunk 3 above)
Comment (penguin0616) [reply modmain.lua]: Good point. 
Done.

**Comment by penguin0616** [comment]
Alright, I think that those issues are solved.  Anything else you want me to change?

**Review by rezecib — APPROVED** [review]

**Comment by rezecib** [comment]
Thanks so much for adding this! Just testing locally and then I'll update the documentation to mention the integration with Insight and do a release on Steam.

**Comment by penguin0616** [comment]
It was my pleasure. :D

**Pull request merged**